lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -960,7 +960,12 @@ def headers(options)
         cookie = options[:cookie] || cookie
         headers['Cookie'] = cookie if cookie
 
-        headers['Idempotency-Key'] = idempotency_key if idempotency_key
+        # Required because Worldpay does not accept duplicate idempotency keys
+        # for different transactions, such as in the case of an authorize => capture flow.
+        if @idempotency_key
+          headers['Idempotency-Key'] = @idempotency_key
+          options[:idempotency_key] = SecureRandom.uuid
```

Comment:
I think this line should be changing the instance variable
```suggestion
          @idempotency_key = SecureRandom.uuid
```

Reply:
fixed!